src/components/feed/AddNode/GuidedConfig.tsx

```diff
@@ -87,7 +87,7 @@ const GuidedConfig = ({
       });
     }
 
-    if (params && count >= params.length) {
+    if (params && params.length !== 0 && count > params.length - 1) {
```

Comment:
The problem is that, the HTTP request which populates the variable `params` happens asynchronously. During the time the HTTP request is happening, the UI renders itself, and _(on a slow internet connection)_ it could render before data is available, so that the UI will display "You cannot add more parameters to this plugin", which is not true, in reality, the UI should be displaying "Plugin parameters are loading..." or some other kind of loading bar.

Could you explain to me how your change fixes the race condition?

Reply:
@jennydaman What I did here is` params.length !== 0` was checking if the length of the params array is not zero which means params have been fetched and `count > params.length - 1` checks if the total number of params added (count) is greater than the total number of params. This way the error shows only when params have been fetched and all params have been added.